superset-frontend/webpack.proxy-config.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable no-console */
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+const fs = require('fs');
+const zlib = require('zlib');
+const path = require('path');
+// eslint-disable-next-line import/no-extraneous-dependencies
+const parsedArgs = require('yargs').argv;
+
+const { supersetPort = 8088, superset: supersetUrl = null } = parsedArgs;
+const backend = (supersetUrl || `http://localhost:${supersetPort}`).replace(
+  '//+$/',
+  '',
+); // strip ending backslash
+const MANIFEST_FILE = path.resolve(
+  __dirname,
+  '../superset/static/assets/manifest.json',
+);
+
+let manifestContent;
+let manifest;
+function loadManifest() {
+  try {
+    const newContent = fs.readFileSync(MANIFEST_FILE);
+    if (!newContent || newContent === manifestContent) return;
+    manifestContent = newContent;
+    manifest = JSON.parse(manifestContent);
+    console.log(`${MANIFEST_FILE} loaded.`);
+  } catch (e) {
+    if (e.code !== 'ENOENT') {
+      console.error('\n>> Error loading manifest file:');
+      console.trace(e);
+    }
+  }
+}
+// make sure the manifest file exists
+fs.mkdirSync(path.dirname(MANIFEST_FILE), { recursive: true });
+fs.closeSync(fs.openSync(MANIFEST_FILE, 'as+'));
+fs.watch(MANIFEST_FILE, loadManifest);
+
+function isHTML(res) {
+  const CONTENT_TYPE_HEADER = 'content-type';
+  const contentType = res.getHeader
+    ? res.getHeader(CONTENT_TYPE_HEADER)
+    : res.headers[CONTENT_TYPE_HEADER];
+  return contentType.includes('text/html');
+}
+
+function toDevHTML(origHtml) {
+  let html = origHtml.replace(
```

Comment:
* could move this block of code below the `loadManifest()` block.
* can use full word for clarity `origHtml` => `originalHtml`

Reply:
I think `origHtml` is more in line with `origRes` used elsewhere in this file, otherwise should I rename `origRes` to `originalResponse`, henceforth `res` to `response`, and `req` to `request` as well?

Reply:
Updated the variable names!